### PR TITLE
Devicelogs

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @dannyroberts sort of :blowfish: but didn't do a great job

did a check on where `DEVICE_LOG_XMLNS` and `http://code.javarosa.org/devicereport` are used and as far as i'm aware isn't almost only used to filter _out_ device logs. so this change shouldn't really impact anything

### CODE REVIEW

File: corehq/apps/users/dbaccessors/__init__.py [hunk 1]
```diff
@@ -1,2 +1,3 @@
 from .all_commcare_users import (get_all_commcare_users_by_domain,
                                  get_user_docs_by_username)
+from .couch_users import get_user_id_by_username
```
Comment (dimagimon) [inline corehq/apps/users/dbaccessors/__init__.py]: F401 'get_user_id_by_username' imported but unused

File: corehq/ex-submodules/couchforms/tests/test_devicelogs.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        self.assertEqual(DeviceReportEntry.objects.count(), 7)
+        first = DeviceReportEntry.objects.first()
+
+        self.assertEqual(first.type, 'resources')
+        self.assertEqual(first.domain, 'test-domain')
+        self.assertEqual(first.msg, 'Staging Sandbox: 54cfe6515fc24e3fafa1170b9a7c2a00')
+        self.assertEqual(first.device_id, '351780060530179')
+        self.assertEqual(
+            first.app_version,
+            'CommCare ODK, version "2.15.0"(335344). App v182. CommCare Version 2.15. Build 335344, built on: October-02-2014'
```
Comment (dimagimon) [inline corehq/ex-submodules/couchforms/tests/test_devicelogs.py]: E501 line too long (126 > 115 characters)

File: corehq/ex-submodules/couchforms/util.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 from __future__ import absolute_import
 import hashlib
 import datetime
+import dateutil.parser
```
Comment (dimagimon) [inline corehq/ex-submodules/couchforms/util.py]: F401 'dateutil' imported but unused

File: corehq/ex-submodules/couchforms/util.py [hunk 4]
```diff
@@ -17,12 +18,16 @@
...
 from corehq.form_processor.interfaces.processor import FormProcessorInterface
 from corehq.util.soft_assert import soft_assert
+from corehq.apps.users.dbaccessors import get_user_id_by_username
+from corehq.apps.users.util import format_username
+from phonelog.models import UserEntry, DeviceReportEntry
 from dimagi.utils.couch.undo import DELETED_SUFFIX
 from dimagi.utils.logging import notify_exception
 from dimagi.utils.mixins import UnicodeMixIn
 from dimagi.utils.couch import LockManager
+from dimagi.utils.parsing import string_to_utc_datetime
```
Comment (dimagimon) [inline corehq/ex-submodules/couchforms/util.py]: F401 'string_to_utc_datetime' imported but unused

**Comment by benrudolph** [comment]
cc: @proteusvacuum

File: corehq/ex-submodules/couchforms/tests/test_devicelogs.py (hunk 2 above)
Comment (benrudolph) [inline corehq/ex-submodules/couchforms/tests/test_devicelogs.py]: ...gonna just roll with it

**Comment by dannyroberts** [comment]
Would a good summary of this PR be something like the following?

> - Stop saving device logs to couch at all; save directly to SQL instead
> - remove the Pillow that was doing the async transfer previously
> - At any point in the future we can go back and delete all device log forms from couch

Just confirming; if so—this looks great! Really psyched to see what kind of performance gains we get from this!

**Comment by benrudolph** [comment]
that's correct. sorry for some reason i left off any sort of summary. in the zone

**Comment by benrudolph** [comment]
also @ctsims this means we won't be saving the form to couch and we won't be saving the form.xml, but the device report will still work

File: corehq/apps/users/tests/test_db_accessors.py [hunk 5]
```diff
@@ -79,3 +84,8 @@ def test_get_all_ids(self):
         self.assertEqual(4, len(all_ids))
         for id in [self.ccuser_1._id, self.ccuser_2._id, self.web_user._id, self.ccuser_other_domain._id]:
             self.assertTrue(id in all_ids)
+
+    def test_get_id_by_username(self):
+        user_id = get_user_id_by_username(self.ccuser_1.username)
+        self.assertEqual(user_id, self.ccuser_1._id)
+
```
Comment (dimagimon) [inline corehq/apps/users/tests/test_db_accessors.py]: W391 blank line at end of file

**Comment by czue** [comment]
looks good! I'm going to wait for a discussion on whether this should be asynchronous before merging. Will be awesome to get this in!

File: corehq/ex-submodules/couchforms/tests/test_devicelogs.py (hunk 2 above)
Comment (snopoke) [inline corehq/ex-submodules/couchforms/tests/test_devicelogs.py]: https://docs.google.com/document/d/1uGE9S-FDSI4ovTLu_fX-AWh2MkHqL8snDNHa1vAtrnM/edit

File: corehq/ex-submodules/couchforms/util.py [hunk 6]
```diff
@@ -531,3 +537,65 @@ def spoof_submission(submit_url, body, name="form.xml", hqsubmission=True,
...
+            msg=log["msg"],
+            # must accept either date or datetime string
+            date=log["@date"],
+            server_date=xform.received_on,
+            app_version=form_data.get('app_version'),
+            device_id=form_data.get('device_id'),
+            username=logged_in_username,
+            user_id=logged_in_user_id,
+        ))
+    DeviceReportEntry.objects.bulk_create(to_save)
```
Comment (snopoke) [inline corehq/ex-submodules/couchforms/util.py]: Feels like this code could live in another file

Comment (benrudolph) [inline corehq/ex-submodules/couchforms/util.py]: makes sense will update

**Comment by dannyroberts** [comment]
Looks good to me!

**Comment by benrudolph** [comment]
ok all comments addressed i believe, just waiting on tests

**Comment by czue** [comment]
woop woop! :racehorse:

**Comment by snopoke** [comment]
awesome!

**Comment by dannyroberts** [comment]
:100:

**Pull request merged**